### ISSUE TITLE
Not able to deploy convoy-nfs catalog.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#4602]
Rancher	v1.1.0-dev1-rc1
Cattle	v0.160.0
User Interface	v1.0.6
Rancher Compose	v0.8.0-rc3

Docker version on host -1.10.3

Deploy convoy-nfs catalog from catalog "Library"/"Rancher Services"

convoy-nfs stack is stuck in "Activating" state 

<img width="1131" alt="screen shot 2016-04-28 at 9 24 39 pm" src="https://cloud.githubusercontent.com/assets/4266958/14908147/dc38ae92-0d87-11e6-85b7-ec6c807a2c54.png">

```
time="2016-04-29T04:18:48Z" level=info msg="Starting rancher-compose-executor" version=v0.8.0-rc3 
time="2016-04-29T04:18:48Z" level=info msg="Initializing event router" workerCount=10 
time="2016-04-29T04:18:48Z" level=info msg="Connection established" 
time="2016-04-29T04:19:31Z" level=info msg="Stack Create Event Received" eventId=706d4316-e553-4805-9d75-d8294cca6183 resourceId=1e382 
time="2016-04-29T04:19:31Z" level=info msg="[convoy-nfs:convoy-nfs]: Adding " eventId=706d4316-e553-4805-9d75-d8294cca6183 resourceId=1e382 
time="2016-04-29T04:19:31Z" level=info msg="[convoy-nfs:convoy-nfs-storagepool]: Adding " eventId=706d4316-e553-4805-9d75-d8294cca6183 resourceId=1e382 
time="2016-04-29T04:19:31Z" level=info msg="[convoy-nfs:]: Creating project " eventId=706d4316-e553-4805-9d75-d8294cca6183 resourceId=1e382 
time="2016-04-29T04:19:31Z" level=info msg="[convoy-nfs:convoy-nfs]: Creating " eventId=706d4316-e553-4805-9d75-d8294cca6183 resourceId=1e382 
time="2016-04-29T04:19:31Z" level=info msg="[convoy-nfs:convoy-nfs-storagepool]: Creating " eventId=706d4316-e553-4805-9d75-d8294cca6183 resourceId=1e382 
time="2016-04-29T04:19:31Z" level=info msg="Creating service convoy-nfs" 
panic: runtime error: index out of range

goroutine 44 [running]:
panic(0xaabfc0, 0xc82000a030)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/docker.volumes(0xc820682a80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc82034ef60, 0x1, ...)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/docker/convert.go:59 +0x2c5
github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/docker.Convert(0xc820682a80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc82034ef60, 0x1, ...)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/docker/convert.go:138 +0x497
github.com/rancher/rancher-compose/rancher.createLaunchConfig(0xc82016ebe0, 0xc8200ca220, 0xa, 0xc820682a80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/rancher/rancher-compose/rancher/configs.go:59 +0x241
github.com/rancher/rancher-compose/rancher.createLaunchConfigs(0xc82016ebe0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/rancher/rancher-compose/rancher/configs.go:18 +0xfe
github.com/rancher/rancher-compose/rancher.(*NormalFactory).config(0xfb8dc8, 0xc82016ebe0, 0x96aca0, 0xc8201ceb10, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher-compose/rancher/normal.go:45 +0x7c
github.com/rancher/rancher-compose/rancher.(*NormalFactory).configAndHash(0xfb8dc8, 0xc82016ebe0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x462ae7, 0x0, 0x0)
	/go/src/github.com/rancher/rancher-compose/rancher/normal.go:21 +0x7c
github.com/rancher/rancher-compose/rancher.(*NormalFactory).Create(0xfb8dc8, 0xc82016ebe0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher-compose/rancher/normal.go:75 +0x6c
github.com/rancher/rancher-compose/rancher.(*RancherService).createService(0xc82016ebe0, 0xc8200ca220, 0x0, 0x0)
	/go/src/github.com/rancher/rancher-compose/rancher/service.go:297 +0x1ef
github.com/rancher/rancher-compose/rancher.(*RancherService).Create(0xc82016ebe0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher-compose/rancher/service.go:73 +0x78
github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project.(*Project).Create.func1.1(0x7f841ff1dcc8, 0xc82016ebe0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project/project.go:181 +0x33
github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project.(*serviceWrapper).Do(0xc8202c15e0, 0xc8204a7aa0, 0x7, 0x8, 0xd1be98)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project/service-wrapper.go:100 +0x202
github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project.(*Project).Create.func1(0xc8202c15e0, 0xc8204a7aa0)
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project/project.go:182 +0x49
created by github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project.(*Project).startService
	/go/src/github.com/rancher/rancher-compose/vendor/github.com/docker/libcompose/project/project.go:345 +0x5a8
```

**Comment by joshwget** [comment]
Looks like the same panic as with the Kubernetes issue. Should be fixed by rancher/rancher-compose#153.

**Comment by sangeethah** [comment]
Tested with v1.1.0-dev1-rc2
Able to  deploy convoy-nfs catalog successfully.
convoy-nfs stack gets activated successfully and storage pool gets created.
Able to start services having data volumes in "convoy-nfs" volume driver.
